### PR TITLE
feat(rust,python): Add `is_local` and `to_local` to categorical namespace

### LINKED ISSUE #10360: Add `to_local` method to categorical namespace
### Problem description

In order to convert our categoricals to the dataframe interchange format, we need to be able to 'localize' a categorical column.

This method would make the physical representation match the indices of the `categories` array.

Example test:

```python
def test_categorical_to_local() -> None:
    with pl.StringCache():
        s1 = pl.Series("a", ["foo"], dtype=pl.Categorical)
        s2 = pl.Series("b", ["bar"], dtype=pl.Categorical)

    assert s1.to_physical().item() == 0
    assert s2.to_physical().item() == 1

    out = s2.cat.to_local()
    assert out.to_physical().item() == 0
    assert out.item() == "bar"
```

Blocker for #10267

### PR DESCRIPTION
**Pull request by stinodego**
Closes #10360 

Changes:
* Add `Series.cat.is_local` to help check whether a series is local or global
* Add `Series/Expr.cat.to_local` to convert a categorical to its local representation

Blocker for #10267

### CODE REVIEW

**Comment by stinodego** [comment]
This is ready for review.

CI failure is due to https://github.com/nushell/reedline/issues/616

File: py-polars/docs/source/reference/expressions/categories.rst [hunk 1]
```diff
@@ -11,3 +11,4 @@ The following methods are available under the `expr.cat` attribute.
 
     Expr.cat.get_categories
     Expr.cat.set_ordering
+    Expr.cat.to_local
```
Comment (ritchie46) [inline py-polars/docs/source/reference/expressions/categories.rst:14]: I don't think this should be an expression. It seems to me  like an implementation detail we only want on the memory buffers, i.e. Series

File: crates/polars-core/src/chunked_array/logical/categorical/mod.rs [hunk 2]
```diff
@@ -48,16 +48,39 @@ impl CategoricalChunked {
...
+        let (physical_map, categories) = match rev_map.as_ref() {
+            RevMapping::Global(m, c, _) => (m, c),
+            RevMapping::Local(_) => return self.clone(),
+        };
+
+        let local_rev_map = RevMapping::Local(categories.clone());
+        // TODO: A fast path can possibly be implemented here:
+        // if all physical map keys are equal to their values,
+        // we can skip the apply and only update the rev_map to local
+        let local_ca = self.logical().apply(|v| *physical_map.get(&v).unwrap());
```
Comment (ritchie46) [inline crates/polars-core/src/chunked_array/logical/categorical/mod.rs:74]: This is my bad. 

Now that I think about it. If the array has got null values, the masked out value might be an incorrect index. So we must use `apply_on_opt` here.

File: py-polars/docs/source/reference/expressions/categories.rst (hunk 1 above)
Comment (stinodego) [reply py-polars/docs/source/reference/expressions/categories.rst:14]: Okay, let's have it on Series only for now!

Done.

File: crates/polars-core/src/chunked_array/logical/categorical/mod.rs (hunk 2 above)
Comment (ritchie46) [reply crates/polars-core/src/chunked_array/logical/categorical/mod.rs:74]: You have to create a categorical that is out of bounds under the mask, not always easy. Can checkout this PR later.

Comment (stinodego) [reply crates/polars-core/src/chunked_array/logical/categorical/mod.rs]: Updated the implementation to use `apply_on_opt` 👍

**Review by ritchie46 — APPROVED** [review]
Looks great!

**Pull request merged**